File: templates/java-webflux/subsystem/user/component/src/main/java/cobalt/user/UserRepository.java
package cobalt.user;

import cobalt.user.model.User;
import org.springframework.stereotype.Repository;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

import java.util.List;

@Repository
class UserRepository {

  static List<User> users = List.of(
    new User(1, "Talia", "Fritsch"),
    new User(2, "Daisy", "Lubowitz"),
    new User(3, "Chaz", "Rolfson"),
    new User(4, "Nelda", "Wolff"),
    new User(5, "Giovanna", "Nader"),
    new User(6, "Jerald", "Romaguera"),
    new User(7, "Faustino", "Nienow")
  );

  Flux<User> findAll() {
    return Flux.fromIterable(users);
  }

  Mono<User> findById(Integer id) {
    final User user = users
      .stream()
      .filter(it -> it.id.equals(id))
      .findFirst()
      .orElse(null);

    if (user != null) {
      return Mono.just(user);
    } else {
      return Mono.empty();
    }
  }

}

File: templates/kotlin-webflux/subsystem/user/component/src/main/kotlin/cobalt/user/UserRepository.kt
package cobalt.user

import cobalt.user.model.User
import org.springframework.stereotype.Repository
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

@Repository
internal class UserRepository {

  companion object {
    private val users = listOf(
      User(id = 1, familyName = "Talia", givenName = "Fritsch"),
      User(id = 2, familyName = "Daisy", givenName = "Lubowitz"),
      User(id = 3, familyName = "Chaz", givenName = "Rolfson"),
      User(id = 4, familyName = "Nelda", givenName = "Wolff"),
      User(id = 5, familyName = "Giovanna", givenName = "Nader"),
      User(id = 6, familyName = "Jerald", givenName = "Romaguera"),
      User(id = 7, familyName = "Faustino", givenName = "Nienow")
    )
  }

  fun findAll(): Flux<User> =
    Flux.fromIterable(users)

  fun findById(id: Int): Mono<User> {
    val user = users.find { it.id == id }
    return if (user != null) Mono.just(user) else Mono.empty()
  }

}
